main.cpp

```diff
@@ -24,408 +22,153 @@
  
  7) make sure it compiles & runs without errors.
  
- 8) use your knowledge of casting to remove any conversion warnings. 
+ 8) use your knowledge of casting to remove any conversion warnings.
  
  Commit your changes by clicking on the Source Control panel on the left, entering a message, and click [Commit and push].
  
 Make a pull request after you make your first commit and pin the pull request link to our DM thread.
-
 send me a DM to check your pull request
-
  Wait for my code review.
  */
 
-
-
-
 #include <iostream>
 
-struct DoubleType;
-struct IntType;
-
 struct FloatType
 {
-    FloatType(float val) : value(new float(val)) {}
-    ~FloatType() { delete value; }
+    FloatType(float val = 0.0f) : m_value(new float(val)) {}
+    ~FloatType()
+	{
+		delete m_value;
+		m_value = nullptr;
+	}
 
-    float* value;
+	operator float() { return *m_value; }
 
-    // primitives
     FloatType& add(float v);
     FloatType& subtract(float v);
     FloatType& multiply(float v);
     FloatType& divide(float v);
-    // FloatType args
-    FloatType& add(const FloatType& ft);
-    FloatType& subtract(const FloatType& ft);
-    FloatType& multiply(const FloatType& ft);
-    FloatType& divide(const FloatType& ft);
-    // DoubleType args
-    FloatType& add(const DoubleType& dt);
-    FloatType& subtract(const DoubleType& dt);
-    FloatType& multiply(const DoubleType& dt);
-    FloatType& divide(const DoubleType& dt);
-    // IntType args
-    FloatType& add(const IntType& it);
-    FloatType& subtract(const IntType& it);
-    FloatType& multiply(const IntType& it);
-    FloatType& divide(const IntType& it);
+	
+private:
+	float* m_value { nullptr };
```

Comment:
- [ ] FIXME: don't use `m_` in variable names.  Why did you change this?  You're just supposed to make it private, not rename it.

Reply:
Cos I saw it used all over the place on learncpp.com and thought it was good practice. 

Here's the related paragraph

> Using the “m_” prefix for member variables helps distinguish member variables from function parameters or local variables inside member functions. This is useful for several reasons. First, when we see an assignment to a variable with the “m_” prefix, we know that we are changing the state of the class instance. Second, unlike function parameters or local variables, which are declared within the function, member variables are declared in the class definition. Consequently, if we want to know how a variable with the “m_” prefix is declared, we know that we should look in the class definition instead of within the function.

So I suppose this would be an instance of the "Hungarian notation"? If so I just saw it on the JUCE coding standard that it's not allowed.